utils/collated_reports.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2025 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import subprocess
+from pathlib import Path
+
+
+DEFAULT_GPU_NAMES = ["mi300", "mi355", "h100", "a10"]
+
+
+def simplify_gpu_name(gpu_name: str, simplified_names: list[str]) -> str:
+    matches = []
+    for simplified_name in simplified_names:
+        if simplified_name in gpu_name:
+            matches.append(simplified_name)
+    if len(matches) == 1:
+        return matches[0]
+    return gpu_name
+
+
+def parse_short_summary_line(line: str) -> tuple[str | None, int]:
+    if line.startswith("PASSED"):
+        return "passed", 1
+    if line.startswith("FAILED"):
+        return "failed", 1
+    if line.startswith("SKIPPED"):
+        line = line.split("[", maxsplit=1)[1]
+        line = line.split("]", maxsplit=1)[0]
+        return "skipped", int(line)
+    if line.startswith("ERROR"):
+        return "error", 1
+    return None, 0
+
+
+def get_paths(p: str, glob_pattern: str | None) -> list[Path]:
+    # Validate path and apply glob pattern if provided
+    path = Path(p)
+    assert path.is_dir(), f"Path {path} is not a directory"
+    if glob_pattern is None:
+        return [path]
+
+    return [p for p in path.glob(glob_pattern) if p.is_dir()]
+
+
+def get_gpu_name(gpu_name: str | None) -> str:
+    # Get GPU name if available
+    if gpu_name is None:
+        try:
+            import torch
+
+            gpu_name = torch.cuda.get_device_name()
```

Comment:
this would not get the result when it is from `.github/workflows/collated-reports.yml` as we are running on ubuntu